internal/pkg/cli/deploy/deploy.go

```diff
@@ -1176,23 +1212,32 @@ func validateAppVersion(appName string, appVersionGetter versionGetter) error {
 	return nil
 }
 
-func validateLBWSRuntime(app *config.Application, envName string, mft *manifest.LoadBalancedWebService, appVersionGetter versionGetter) error {
-	if app.Domain == "" && mft.HasAliases() {
-		log.Errorf(aliasUsedWithoutDomainFriendlyText)
-		return errors.New("alias specified when application is not associated with a domain")
+func validateLBWSRuntime(app *config.Application, env *config.Environment, mft *manifest.LoadBalancedWebService, appVersionGetter versionGetter, validator aliasCertValidator) error {
+	if !mft.HasAliases() {
+		return nil
+	}
+	if env.HasImportedCerts() {
+		aliases, err := mft.RoutingRule.Alias.ToStringSlice()
+		if err != nil {
+			return fmt.Errorf("convert aliases to string slice: %w", err)
+		}
+		if err := validator.ValidCertificatesAliases(aliases, env.CustomConfig.ImportCertARNs); err != nil {
+			return fmt.Errorf("validate aliases against the imported certificate for env %s: %w", env.Name, err)
+		}
+		return validateLBSvcAlias(mft.NLBConfig.Aliases, app, env.Name)
 	}
-
 	if app.RequiresDNSDelegation() {
```

Comment:
Totally unrelated thought: we can maybe change this method name to `app.DNSDelegated` 🤔  Feels like a better description of its intention.